pkg/virtctl/imageupload/imageupload.go

```diff
@@ -107,15 +112,15 @@ func NewImageUploadCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
 	}
 	cmd.Flags().BoolVar(&insecure, "insecure", insecure, "Allow insecure server connections when using HTTPS.")
 	cmd.Flags().StringVar(&uploadProxyURL, "uploadproxy-url", "", "The URL of the cdi-upload proxy service.")
-	cmd.Flags().StringVar(&pvcName, "pvc-name", "", "The destination PVC.")
+	cmd.Flags().StringVar(&pvcName, "pvc-name", "", "The destination DataVolume/PVC name.")
```

Comment:
This does not look right. How can `pvc-name` be the datavolume or the PVC?

Reply:
PVCs created by the datavolume controller always have the same name as the DataVolume (can't be different).   Also, I still want it to be possible to upload to PVCs created out of band (so can't get rid of `pvc-name`).  Finally, I didn't want to break any existing scripts that use `virtctl image-upload`.  I thought about adding another arg called `datavolume-name` that is basically an alias to `pvc-name`.  But it seemed confusing to me.  Oh yeah, and it is the PVC name that will be used in any VM/VMI definition later on.  So, for all those reasons, I think it is best to keep `pvc-name`. 

What makes the most sense to you?

Reply:
> PVCs created by the datavolume controller always have the same name as the DataVolume (can't be different)

I was not aware of this and it sounds dangerous, if this is not somehow verified via owner references or such (but I guess it is verified?).

> What makes the most sense to you?

Since providing a name is mandatory, I would suggest a similar semantic to what we have on e.g. `kubectl expose`: `virtctl upload pvc/myvmc ...` and `virtctl upload dv/mydv`.

We could deprecate `--pvc-name` and maybe remove it later.

Would that make sense?

Reply:
@rmohr please take another look